src/replay_stage.rs

```diff
@@ -191,6 +191,7 @@ impl ReplayStage {
         let to_leader_sender = to_leader_sender.clone();
         let subscriptions_ = subscriptions.clone();
 
+        let entry_height = bank_forks_info[0].entry_height;
```

Comment:
This can be cleaned up a ton. 
1) Return `entry_height` as a new entry in the `(bank, last_entry_id)` tuple since it's that code block that makes the decision to randomly pick `bank_forks_info[0]` -- keep the badness scoped.
2) The block immediately after that assigns `let mut current_blob_index = {` can also be cleaned up now.   If `entry_height` is really `current_blob_index` then call it `current_blob_index` from the start

Reply:
oops, that was really messy. Thanks!